Rewrite "what is ngrok" page to match latest product story (#2192)
ngrok/ngrok-docs: what-is-ngrok.mdx

> description: ngrok is a globally distributed reverse proxy that secures, protects, and accelerates your applications and network services.

Reviewer: We should update this, too!

---

ngrok is developer infrastructure that routes and secures traffic to your apps, APIs, and AI models. 

> It provides agnostic solutions for managing connectivity between all of the tools and platforms you use to put your services online. 

Reviewer: ```suggestion
We give you ways to manage connectivity between all of the tools and platforms you use to put your services online, no matter where they're deployed or what network they're on. 
```

> ngrok's core products are Gateway, AI Gateway, and share localhost.

Reviewer: Hmm, I'm not sure what I think about the capitalization here. Usually I'm against, but I think **Share localhost**, looks a touch better.

```suggestion
ngrok's core products are Gateway, AI Gateway, and Share localhost.
```

Key tools and services like the ngrok agent, Traffic Policy, and Traffic Inspector are shared across products.

## Gateway

> ngrok Gateway is all-in-one cloud networking platform that secures, transforms, and routes traffic to all your services, no matter where they run.

Author: ```suggestion
ngrok Gateway is an all-in-one cloud networking platform that secures, transforms, and routes traffic to all your services, no matter where they run.
```

> As opposed to cloud-provider gateways which are locked to their own cloud, ngrok works the same in AWS, Azure, GCP, on-prem, and your customer's networks: same hostname, same Traffic Policy, same observability. 

Reviewer: Maybe something like this?

```suggestion
As opposed to cloud provider gateways, which are locked to services running in their networks, ngrok works the same in AWS, Azure, GCP, on-prem, and your customer's networks. That means you get one cloud-agnostic way to add public/private connectivity to any service, transform traffic, enforce authentication, or observe traffic.
```

If your services span environments, ngrok is one gateway in front of all of them instead of one per cloud.

## AI Gateway

With the ngrok AI Gateway, you can route, secure, and manage traffic to any LLM—cloud or local—on one unified platform.
Use OpenAI, Anthropic, and more without signing up for each provider—ngrok manages the provider keys for you.
Key features include automatic failover, load balancing, and observability.

## share localhost

> Securely expose a web server on localhost to the internet, even from behind a NAT or firewall, with secure tunnels.

Reviewer: ```suggestion
Securely expose a web server, or any other service you're running on `localhost`, to the internet, even from behind a NAT or firewall, with secure tunnels.
```

> ngrok assigns a public endpoint that forwards incoming requests to your local server over an encrypted connection. 

Reviewer: I wonder if this could mention some primary use cases: demos without deploying, testing webhooks, testing mobile apps, letting ChatGPT/Claude/etc "talk" to local MCP servers
